src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,201 @@
+        }
+
+        sendTelemetryEvent(Telemetry.PandasNotInstalled);
+        await this.installMissingDependencies(options);
+    }
+
+    private async getVersionOfPandas(options: IDataViewerDependencyServiceOptions): Promise<SemVer | undefined> {
+        const kernel = (options as IDataViewerDependencyServiceOptionsWithKernel).kernel;
+        const command = kernel ? kernelGetPandasVersion : debuggerGetPandasVersion;
+
+        try {
+            const outputs = await this.executeSilently(command, options);
+            return outputs.map((text) => (text ? parseSemVer(text.toString()) : undefined)).find((item) => item);
+        } catch (e) {
+            traceWarning(DataScience.failedToGetVersionOfPandas(), e.message);
+            return;
+        }
+    }
+
+    private async installMissingDependencies(options: IDataViewerDependencyServiceOptions): Promise<void> {
+        sendTelemetryEvent(Telemetry.PythonModuleInstall, undefined, {
+            action: 'displayed',
+            moduleName: ProductNames.get(Product.pandas)!
+        });
+
+        const kernel = (options as IDataViewerDependencyServiceOptionsWithKernel).kernel;
+
+        let selection: string | undefined;
+        if (kernel) {
+            selection = this.isCodeSpace
+                ? Common.install()
+                : await this.applicationShell.showErrorMessage(
+                      DataScience.pandasRequiredForViewing(),
+                      { modal: true },
+                      Common.install()
+                  );
+        } else {
+            // TODO: Installing pandas does not work with the debugger.
+            // It takes a long time, then it breaks with:
+            // error: Command \"clang -Wno-unused-result -Wsign-compare -Wunreachable-code...
+            await this.applicationShell.showErrorMessage(DataScience.pandasRequiredForViewing());
+        }
+
+        let commands: string[];
+        if (kernel) {
+            commands = [`${this.kernelPackaging(kernel)} install pandas`];
+        } else {
+            commands = debuggerInstallPandas;
+        }
+
+        if (selection === Common.install()) {
+            try {
+                await this.executeSilently(commands, options);
+                sendTelemetryEvent(Telemetry.UserInstalledPandas);
+            } catch (e) {
+                sendTelemetryEvent(Telemetry.FailedToInstallPandas);
+                throw new Error(DataScience.failedToInstallPandas());
+            }
+        } else {
+            sendTelemetryEvent(Telemetry.UserDidNotInstallPandas);
+            throw new Error(DataScience.pandasRequiredForViewing());
+        }
+    }
+
+    private async executeSilently(
+        commands: string[],
+        options: IDataViewerDependencyServiceOptions
+    ): Promise<(string | undefined)[]> {
+        const kernel = (options as IDataViewerDependencyServiceOptionsWithKernel).kernel;
```

Comment:
This doesn't look right.
Dependency service shouldn't take a dependency on a variable provider.
At this point, all we need to do is:
* Check if we have an active debug session
* If we have a debug session, then use the debug sessions `evaluate` method to run some aribtrary code & get the desired output from there.

Today we check in other places whther we have an active Debug session using `IDebugService.activeDebugSession` and if that is not undefined then we know we have a debug session.
Next you can use `IDebugService.activeDebugSesion.customRequest` method to send a custom request.

Reply:
- To use activeSession.customRequest we need a frame Id, and the frame Id of the variable in the Python debugger used to show the data viewer does not work. It runs the code and shows to succeed even in the logs, but it does not yield any value, therefore that frameId is not useful, and no frameId doesn’t work either.
- The debugger cannot currently be used to install pandas, the installation doesn’t finish and the debugger becomes unusable afterwards.

Reply:
Please see my comments here https://github.com/microsoft/vscode-jupyter/issues/10605#issuecomment-1168183711

Again, I don't think we should expose `evaluate` method onn a vairbla provider class for anyone to use.

Reply:
> The debugger cannot currently be used to install pandas, the installation doesn’t finish and the debugger becomes unusable afterwards.

We can get the path to th eexecutable using `import sys;sys.executable` and use the old code to install using the interpreter information.

Reply:
> ctiveSession.customRequest we need a frame Id, and the frame Id of the variable 

Not following your comments here.
Are you saying that frameId is useless or it is required. 
Looking at the code you have exposed `frameId` as an optional argument in some new methods. & if its not used, then I dont see the need to expose such an argument to methods.

>ut it does not yield any value, therefore that frameId is not useful, and no frameId doesn’t work either.

Have you tried a frame id of `0`?

Reply:
Regarding frameId:

- The frameId of the variable coming from the debugger causes the results to be empty.
- Frame Id 0 doesn’t work either, I can show you on a call.
- One other frameId worked for Rich and I but essentially we could only get hold of it by using the evaluate method from IJupyterVariables (If I understood him correctly. It does work — when getting the version at the very least, then the installation has another challenge but the installation begins to work, just doesn’t finish).

Reply:
I did use `sys.executable` [here](https://github.com/microsoft/vscode-jupyter/pull/10604#discussion_r908347095) and it (later, in the middle of the installation) causes a major issue in the web. I believe that you mean only to extract the interpreter path and use the existing interpreter code then, if that’s correct I need to investigate how to make that code work on the web.

Reply:
>  Python debugging doesn't quite work in the web)

Exactly, hence the only time we're debugging is when we launch the debugger for IW or notebooks & we know exactly what the IKernel is in that case.